text/entities/0264-resource-and-entities.md

```diff
@@ -0,0 +1,284 @@
+
+## Datamodel Changes
+
+Given our desired design and algorithms for detecting, merging and manipulating Entities, we need the ability to denote how entity and resource relate. These changes must not break existing usage of Resource, therefore:
+
+- The Entity model must be *layered on top of* the Resource model.  A system does not need to ineract with entities for correct behavior.
+- Existing key usage of Resource must remain when using Entities, specifically navigationality (see: [OpenTelemetry Resources: Principles and Characteristics](https://docs.google.com/document/d/1Xd1JP7eNhRpdz1RIBLeA1_4UYPRJaouloAYqldCeNSc/edit))
+- Downstream components should be able to engage with the Entity model in Resource.
+
+The following changes are made:
+
+### Resource
+
+| Field | Type | Description | Changes |
+| ----- | ---- | ----------- | ------- |
+| schema_url | string | The Schema URL, if known. This is the identifier of the Schema that the resource data  is recorded in.  This field is deprecated and should no longer be used. | Will be deprecated |
+| dropped_attributes_count |  integer | dropped_attributes_count is the number of dropped attributes. If the value is 0, then no attributes were dropped. | Unchanged |
+| attributes | repeated KeyValue | Set of attributes that describe the resource.<br/><br/>Attribute keys MUST be unique (it is not allowed to have more than one attribute with the same key).| Unchanged |
+| entities | repeated ResourceEntityRef | Set of entities that participate in this Resource. | Added |
+
+The DataModel would ensure that attributes in Resource are produced from both the identifying and descriptive attributes of Entity.  This does not mean the protocol needs to transmit duplicate data, that design is TBD.
+
+### ResourceEntityRef
+
+The entityref data model, would have the following changes from the original [entity OTEP](https://github.com/open-telemetry/oteps/blob/main/text/entities/0256-entities-data-model.md) to denote references within Resource:
+
+| Field | Type | Description | Changes |
+| ----- | ---- | ----------- | ------- |
+| schema_url | string | The Schema URL, if known. This is the identifier of the Schema that the entity data  is recorded in. To learn more about Schema URL ([see docs](https://opentelemetry.io/docs/specs/otel/schemas/#schema-url)) | added |
+| type | string | Defines the type of the entity. MUST not change during the lifetime of the entity. For example: "service" or "host". This field is required and MUST not be empty for valid entities. | unchanged |
+| identifying_attributes_keys | repeated string | Attribute Keys that identify the entity.<br/>MUST not change during the lifetime of the entity. The Id must contain at least one attribute.<br/><br/>These keys MUST exists in Resource.attributes.<br/><br/>Follows OpenTelemetry common attribute definition. SHOULD follow OpenTelemetry semantic conventions for attributes.| now a reference |
```

Comment:
Listing just the keys complicates operations with entities (to find the value of an entity key you have to do 2 lookups instead of 1). Why not list the full key/value pairs?

Reply:
that's listed below in alternatives considered.

Reply:
We discussed this offline. I think keeping it as key/value is more desirable. We have tools in many places that know how to deal with key/value lists. We don't have any tooling that understands key indexes.

Reply:
Why not move that link completely to the entity schema? It does not make sense to ship that information with every log line that we send around in our observability stack.